apps/3d_rec_framework/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ else(NOT VTK_FOUND)
 endif(NOT VTK_FOUND)
 
 # OpenNI found?
-if(NOT OPENNI_FOUND OR NOT BUILD_OPENNI)
+if(NOT OPENNI_FOUND OR NOT WITH_OPENNI)
```

Comment:
I would propose to set `OPENNI_FOUND` if `WITH_OPENNI` is not set. That way the semantic is clear and we don't have to check both variables all the time.